### DISCUSSION TITLE
Splitting FAIMS .raw Files with Multiple CVs

### DISCUSSION THREAD

**Discussion by alecksphillips** [vdemichev/DiaNN#138] [Q&A]
Hi,

I'm attempting to analyse some FAIMS data that was acquired with two CVs. In the discussion of issue #67 it was suggested that this is possible by splitting the .raw files into files with one CV per file. I've done this by splitting each file into two mzXML files using [this](https://github.com/PNNL-Comp-Mass-Spec/FAIMS-MzXML-Generator) tool, which I've then converted to mzML using MSConvert (using the settings suggested in the DIA-NN manual).

I've then tried to convert these mzML files to .dia format using DIA-NN. Only the files with one of the CVs are able to be converted, the other set of files with the other CV cause DIA-NN to halt without outputting any .dia file, example log below:

```
diann.exe --f "C:\Users\sgaphill\Data\file1.mzML
" --convert --threads 6 --verbose 1
DIA-NN 1.8 (Data-Independent Acquisition by Neural Networks)
Compiled on Jun 28 2021 14:55:31
Current date and time: Mon Jul 12 12:51:40 2021
CPU: GenuineIntel Intel(R) Xeon(R) CPU E5-1620 v3 @ 3.50GHz
SIMD instructions: AVX AVX2 FMA SSE4.1 SSE4.2 
Logical CPU cores: 8
MS data files will be converted to .dia format
Thread number set to 6

1 files will be processed
[0:00] Loading run C:\Users\sgaphill\Data\file1.mzML

DIA-NN exited
```

Is there another method for splitting FAIMS files that is compatible with DIA-NN?

Thanks

**Comment by vdemichev** [comment]
Hi, sorry, don't know, haven't tried this myself. Could you maybe share the file that cannot be read (email me a link), and I will take a look?

Best,
Vadim

**Comment by weiclav** [comment]
Hi there,

try to use directly those mzXML files generated by the splitting tool mentioned, we were using them in 1.7.x versions without any problem, no need for further conversion using msconvert.

Take care,
David

**Reply by alecksphillips** [reply]
Hi David,

Unfortunately it doesn't work with these mzXML files (tried with v1.8 and v1.7.16), DIA-NN complains that they're an invalid format:
```
...
WARNING: skipping C:\Users\sgaphill\Data\splitmzXML-40.mzXML - invalid raw MS data format
WARNING: skipping C:\Users\sgaphill\Data\splitmzXML-60.mzXML - invalid raw MS data format

0 files will be processed
Finished
```

**Reply by vdemichev** [reply]
Yes, that's expected with mzXML. But if I modify the DIA-NN code a bit to allow .mzXML, it still crashes the same way it does with .mzML. Apparently some incompatibility between the output of the FAIMS splitting tool and DIA-NN, on these particular runs. I will look into this further, but this might take time.

**Reply by alecksphillips** [reply]
Thanks Vadim. If I can give you any more information to help, please let me know.

Best Wishes,
Alex

**Reply by weiclav** [reply]
I have double checked it now and I have mixed this with DDA data... very sorry for the confusion guys

good luck with it

take care

**Comment by DavidRossHall** [comment]
> Yes, that's expected with mzXML. But if I modify the DIA-NN code a bit to allow .mzXML, it still crashes the same way it does with .mzML. Apparently some incompatibility between the output of the FAIMS splitting tool and DIA-NN, on these particular runs. I will look into this further, but this might take time.

We ran into a similar issue with *ThermoFAIMStoMzML*, wherein the DIA-NN analysis of the outputted files would return the following error: 

```
[0:13] First pass: generating a spectral library from DIA data
[0:13] File #1/3
[0:13] Loading run SomeFile_-40.mzML
No MS2 spectra: aborting
ERROR: cannot load the file, skipping
[0:13] 0 library precursors are potentially detectable
```

Apparently DIA-NN expects sequentially numbered MS scan numbers, whereas [*ThermoFAIMStoMzML*](https://github.com/PNNL-Comp-Mass-Spec/Thermo-FAIMS-to-MzML) would carry the original scan numbers to the split files. Basically you’d have your `.raw` file with scans numbered like: CV1 Scan 1, CV1 Scan 2… CV1 Scan 10, CV2 Scan 11… CV2 Scan 20, CV1 Scan 21… and so on. When split using *Thermo Freestyle*, the scans are renumbered to CV1 file scan numbers go 1, 2, … and CV2 file scan numbers go 1, 2, … However, with ThermoFAIMStoMzML, the CV1 file scan numbers would go 1, 2… 10, 21, 21… and the CV2 file scan numbers would go 11, 12…31, 32…. 

@alchemistmatt has released [*ThermoFAIMStoMzML* Release 1.0.8783](https://github.com/PNNL-Comp-Mass-Spec/Thermo-FAIMS-to-MzML/releases/tag/v1.0.8783) with a new option (`/RenumberScans`) that  renumbers the scans in the output files so that the first scan is always scan 1 and there are no scan gaps. The resulting `.mzml` files can be run by DIA-NN 1.8.1.

**Reply by vdemichev** [reply]
DIA-NN uses mstoolkit https://github.com/mhoopmann/mstoolkit as mzML & raw reader, I guess we will implement a custom raw reader in the near future.
